gallery.cpp

```diff
@@ -73,3 +76,20 @@ Gallery::Gallery(QWidget *parent) : QWidget(parent)
     progressIndicatorSpinning->animate();
     layout->addWidget(progressIndicatorSpinning);
 }
+
+bool Gallery::eventFilter(QObject *watched, QEvent *event)
+{
+    if (watched == searchField) {
+        if (event->type() == QEvent::KeyPress) {
+            QKeyEvent* e = static_cast<QKeyEvent*>(event);
+            switch (e->key()) {
+                case Qt::Key_Return:
+                    QMessageBox::information(this, "Key event received",
+                        "Key event \"" + QKeySequence(e->key()).toString() +
+                        "\" received on searchField");
```

Comment:
I wasn't sure how to make visible that my change was working.
Thinking again about this, I think having a simple qDebug() message would be nicer: still visible, but less intrusive.
Or we can simply remove this test code.
What do you think?